pages/DSNP/DSNP-Announce.md

```diff
@@ -0,0 +1,62 @@
+# DSNP Announce
+## Specification Status
+
+| Version | Status |
+---------- | ---------
+| 0.1     | Draft |
+
+## Purpose
+1. Describe the method to announce [DSNP Messages](/DSNP/DSNP-Messages) to the blockchain.
+1. Provide Solidity code examples (pending final contract)
+1. Facilitate use of SDK and locating of on-chain data
+
+## Assumptions
+* Chain messages are on Ethereum
+* Message data is posted via [Ethereum log events](https://medium.com/mycrypto/understanding-event-logs-on-the-ethereum-blockchain-f4ae7ba50378)
+* Signature algorithm is [secp256k1](https://en.bitcoin.it/wiki/Secp256k1). This allows the use `ecreover`
+  to get public keys. A public key also need not be included in a log event for ease of validation.
+* content hashes are created via the same [keccak-256 hashing algorithm](https://en.wikipedia.org/wiki/SHA-3) used by Ethereum.
+
+## DSNP Announce
+
+### DSNPMessage Event
+
+The DSNPMmessage event is a standard event topic to be used for all DSNP Messages.
+
+```solidity
+interface DSNPAnnounce {
+    event DSNPMessage(address from, uint16 dsnpType, bytes32 data);
+}
+```
+
+| field | description | type |
+|-------|-------------|------|
+| from | Social Identity From Address | address / uint160 / bytes20 |
```

Comment:
1. Should we have this here?
2. Should we name it something different?
3. Should it use the Ethereum address type or do we want to make it generic bytes?

Reply:
I happen to like the convention either of `from` or `fromAddr` as a name.    

If it's generic bytes, then how does it get mapped to a social identity contract?

Reply:
I tried out the name `dsnpFrom` and standardized the prefixes. Thoughts?

In Solidity address, uint160, bytes20 are all synonyms. The connection would likely come as part 2. The identity interface likely will be required to implement https://eips.ethereum.org/EIPS/eip-1271 that would allow for verification at the level of the announcing contract and the end user.

Likely that will also require that the signature (bytes32) be included as a core log level data point.